services/src/main/java/org/keycloak/protocol/saml/mappers/RoleNameMapper.java

```diff
@@ -91,17 +91,15 @@ public String mapName(ProtocolMapperModel model, RoleModel roleModel) {
         RoleContainerModel container = roleModel.getContainer();
         ClientModel app = null;
         if (container instanceof ClientModel) {
-            app = ((ClientModel) container);
+            app = (ClientModel) container;
         }
         String role = model.getConfig().get(ROLE_CONFIG);
         String newName = model.getConfig().get(NEW_ROLE_NAME);
-        String appName = null;
         int scopeIndex = role.indexOf('.');
```

Comment:
I think the check whether role contains '.' is not necessary anymore. If the app is not null and the role doesn't startsWith clientId + "." null will be returned. Otherwise, the PR looks good. Great work with tests, thank you.